packages/redux/src/sharedWishlists/selectors.ts

```diff
@@ -0,0 +1,197 @@
+  );
+
+/**
+ * Retrieves the error state of the current user's shared wishlist.
+ *
+ * @example
+ * ```
+ * import { getSharedWishlistError } from '@farfetch/blackout-redux/sharedWishlists';
+ *
+ * const mapStateToProps = state => ({
+ *     error: getSharedWishlistError(state)
+ * });
+ * ```
+ *
+ * @param state - Application state.
+ *
+ * @returns Error information, `undefined` if there are no errors.
+ */
+export const getSharedWishlistError = (state: StoreState) =>
+  fromSharedWishlistReducer.getError(
+    state.sharedWishlist as SharedWishlistState,
+  );
+
+/**
+ * Retrieves the loading status of the shared wishlist.
+ *
+ * This status is affected by the loading of the shared wishlist itself, as well as any
+ * "add" operation.
+ *
+ * @example
+ * ```
+ * import { isSharedWishlistLoading } from '@farfetch/blackout-redux/sharedWishlists';
+ *
+ * const mapStateToProps = state => ({
+ *     isLoading: isSharedWishlistLoading(state)
+ * });
+ * ```
+ *
+ * @param state - Application state.
+ *
+ * @returns Loading status of the wishlist.
+ */
+export const isSharedWishlistLoading = (state: StoreState) =>
+  fromSharedWishlistReducer.getIsLoading(
+    state.sharedWishlist as SharedWishlistState,
+  );
+
+/**
+ * Retrieves a specific shared whishlist by its id, with all properties populated (ie, the
+ * product).
+ *
+ * @example
+ * ```
+ * import { getSharedWishlist } from '@farfetch/blackout-redux/sharedWhislists';
+ *
+ * const mapStateToProps = (state, { sharedWhishlists: { id } }) => ({
+ *     sharedWhishlist: getSharedWishlist(state, id)
+ * });
+ * ```
+ *
+ * @param state     - Application state.
+ * @param sharedWhishlistId - Numeric identifier of the shared whishlist.
+ *
+ * @returns - Shared Whislist item entity for the given id.
+ */
+export const getSharedWishlist: (
+  state: StoreState,
+  sharedWhishlistId: SharedWishlist['id'],
+) => SharedWishlistEntity = createSelector(
+  [
+    (state: StoreState, sharedWhishlistId: SharedWishlist['id']) =>
+      getEntityById(
+        state,
+        'sharedWishlists',
+        sharedWhishlistId,
+      ) as SharedWishlistEntity,
+  ],
+  (sharedWhishlist): SharedWishlistEntity => ({
+    ...sharedWhishlist,
+  }),
+);
+
+/**
+ * Retrieves a specific shared whishlist item by its id, with all properties populated (ie, the
+ * product).
+ *
+ * @example
+ * ```
+ * import { getSharedWishlistItem } from '@farfetch/blackout-redux/sharedWhislists';
+ *
+ * const mapStateToProps = (state, { sharedWhishlistItem: { id } }) => ({
+ *     sharedWhishlistItem: getSharedWishlistItem(state, id)
+ * });
+ * ```
+ *
+ * @param state     - Application state.
+ * @param sharedWhishlistItemId - Numeric identifier of the shared whishlist item.
+ *
+ * @returns - Shared Whislist item entity for the given id.
+ */
+export const getSharedWishlistItem: (
+  state: StoreState,
+  sharedWhishlistItemId: SharedWishlistItem['id'],
+) => SharedWishlistItemHydrated = createSelector(
```

Comment:
This should not be using createSelector because it will be called by the `getSharedWishlistItems` for each wishlist item in the wishlist which would invalidate the cached response from createSelector. You can use the same approach used for the getWishlistItem selector in this PR: https://github.com/Farfetch/blackout/pull/601/files#diff-1ae78d725a124b5b144b81d9b603c32d1f138581bf6d8689c831e7c68a444675

Reply:
createSelector removed here 👍